packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts

```diff
@@ -152,10 +156,21 @@ export async function askResourceGroupInfo(
     .filter((item) => item.name)
     .map((item) => [item.name, item.location] as [string, string]);
 
-  // TODO: call Azure API directly to list all locations because ARM SDK does not wrap this API.
-  // And then filter by the 'resourceGroup' resource provider.
-  // https://github.com/microsoft/vscode-azuretools/blob/cda6548af53a1c0f538a5ef7542c0eba1d5fa566/ui/src/wizard/LocationListStep.ts#L173
-  const availableLocations = ["East US", "West US"];
+  const credential = await ctx.azureAccountProvider!.getAccountCredentialAsync();
+  const subscriptionClient = new SubscriptionClient(credential as any);
+  const askSubRes = await ctx.azureAccountProvider!.getSelectedSubscription(true);
+  const listLocations = await subscriptionClient.subscriptions.listLocations(
+    askSubRes!.subscriptionId
+  );
+  const locations = listLocations.map((item) => item.displayName);
+  const providerData = await rmClient.providers.get(MsResources);
+  const resourceTypeData = providerData.resourceTypes?.find(
+    (rt) => rt.resourceType?.toLowerCase() === ResourceGroups.toLowerCase()
+  );
+  const resourceLocations = resourceTypeData?.locations;
+  resourceLocations?.filter((item) => locations.includes(item));
```

Comment:
Looks the usage is not correct, as it will return a new array instead of updating current array.

Reply:
Fixed.